src/lib.rs

```diff
@@ -177,6 +175,9 @@ impl Game {
             Some(c) => *c,
         };
 
+        let mut killed_player: Option<&mut Player> = None;
```

Comment:
ah, nice!

I'm pleasantly surprised that the borrow checker allows this!

Reply:
I'm stumbling my way through the dark. I guess I don't know what to expect from the borrow checker at all.